sdk/tables/azure-data-tables/tests/test_encoder.py

```diff
@@ -808,15 +1178,13 @@ def test_encoder_upsert_entity_basic(self, tables_storage_account_name, tables_p
             test_entity = {"PartitionKey": "PK", "RowKey": "RK'@*$!%"}
             expected_entity = {
                 "PartitionKey": "PK",
-                "PartitionKey@odata.type": "Edm.String",
-                "RowKey": "RK'@*$!%",
-                "RowKey@odata.type": "Edm.String",
+                "RowKey": "RK'@*$!%", # this will fail as the encoded RowKey will be "RK''@*$!%"
```

Comment:
Aaah good catch!
I think the issue here is that for update/upsert we actually want to remove the PK and RK from the dict entirely. If we do that this should work properly (however the `_check_backcompat` will fail)

Reply:
I think in both update and upsert entity, we should `pop` PK and RK before calling `encode_entity`.